src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -93,10 +94,18 @@ public class XpathRegressionTest extends AbstractModuleTestSupport {
     );
 
     private static final Set<String> SIMPLE_CHECK_NAMES = getSimpleCheckNames();
-    private static final Map<String, String> ALLOWED_DIRECTORY_AND_CHECKS =
-        getAllowedDirectoryAndChecks();
 
+    /**
+     * Directory containing the corresponding test file.
+     */
+    private static final Map<String, String> DIR_AND_TEST = getAllowedDirectoryAndChecks();
     private static final Set<String> INTERNAL_MODULES = getInternalModules();
+    private static final String JAVA = ".java";
+    private static final Pattern INPUT_XPATH = Pattern.compile("^InputXpath(.+)\\" + JAVA + "$");
```

Comment:
This approach ensures **minimum redundancy (DRY)** while keeping the structure flexible for adding **JavaDoc-style documentation** if needed. Since the content is specific, JavaDoc can provide **proper context information** when required.

Reply:
>while keeping the structure flexible for adding JavaDoc-style documentation if needed. Since the content is specific, JavaDoc can provide proper context information when required.

our religion is: code should read quickly and easily without Javadoc. all jvadoc that we have is payment for following a another religion "eat food we produce" (All checks are applied to checksyle code base).

Reply:
yes we have same opinion that code should speak for itself. eating the own dog food is always a good design approach.